tracing-tracy/src/lib.rs

```diff
@@ -148,64 +150,93 @@ where
     F: for<'writer> FormatFields<'writer> + 'static,
 {
     fn on_new_span(&self, attrs: &Attributes<'_>, id: &Id, ctx: Context<'_, S>) {
-        if let Some(span) = ctx.span(id) {
-            let mut extensions = span.extensions_mut();
-            if extensions.get_mut::<FormattedFields<F>>().is_none() {
-                let mut fields = FormattedFields::<F>::new(String::with_capacity(64));
-                if self.fmt.format_fields(fields.as_writer(), attrs).is_ok() {
-                    extensions.insert(fields);
-                }
+        let Some(span) = ctx.span(id) else {
+            return;
+        };
```

Comment:
I… feel that in cases like these, where `if-let` does not already have a diverging `else` branch, use of `let-else` makes code readability a lil’ bit worse. Even if the semicolon from `return;` were removed so that `rustfmt` collapsed this statement into a single line… it’d be an improvement over the current suggestion, but it is still much easier to notice `if let` than it is to notice the `else { ... }`.

Reply:
Hmmm, yeah this is personal preference. I'm used to early returns and prefer avoiding nesting, but I'm happy to revert if you'd like to keep the old way.

Reply:
I don’t particularly mind this change much as long as `else { return; };` becomes `else { return };` and is formatted accordingly.

Reply:
Sounds good, done!